Lesmis content pages

tcaiger opened the pull request:
### Issue #: [2497](https://github.com/beyondessential/tupaia-backlog/issues/2497) LESMIS Content Pages

### Changes:

- Refactor breadcrumbs
- Add Content Pages

---

### Screenshots:
See issue

tcaiger commented on packages/lesmis/src/views/PageView.js:
```diff
@@ -0,0 +1,61 @@
+  const breadcrumbs = [{ name: title, code: urlSegment }];
+
+  return (
+    <MuiThemeProvider theme={contentPageTheme}>
```
In the designs the typography has it's own set of styles for content pages so I have added a theme to model that.

edmofro replied:
Good call, though I wonder if intentional by @David-G-UX?

tcaiger commented on packages/lesmis/src/constants/page.js:
```diff
@@ -0,0 +1,44 @@
+/*
```
This is obviously pretty basic and wouldn't scale that well if the complexity of the pages were to increase but given the simplicity of the current content pages, I think it works pretty well.

edmofro commented on packages/lesmis/src/theme/contentPageTheme.js:
```diff
@@ -0,0 +1,75 @@
+/*
```
Would be nice if there were a way to "merge modifications" from one theme into a base theme (this has some overlap with the main LESMIS theme)

edmofro commented on packages/lesmis/src/views/PageView.js:
```diff
@@ -0,0 +1,61 @@
+
+export const PageView = ({ content }) => {
+  const { title, body, urlSegment } = content;
+  const breadcrumbs = [{ name: title, code: urlSegment }];
```
Would remove the `code` here, I don't think it's used for anything? Will just need to add a condition check in the breadcrumbs component to not show an entity link if it's the last in the list _or_ has no code

edmofro reviewed (changes requested):
Just one slightly awkward feeling about:
- The breadcrumbs referring to "entity" all over the show where they're now being used for non-entity pages
- The urlSegment for '/about' and '/contact' being repeated in two quite distant places

But I think just allowing the Breadcrumbs to take in "title only" pages, that don't include an entity code, will solve both of those qualms

tcaiger commented:
@edmofro - This is ready for re-review. I have refactored the `breadcrumbs` to use name and urlSegment and removed the concept of entityCode from them and the `useBreadcrumbs` hook us now called `useEntityBreadcrumbs` to reflect their entity context. I also tightened up the `PageContentTheme` a bit to re-use all the properties from the main them apart from typography settings.

edmofro commented on packages/lesmis/src/components/Breadcrumbs.js:
```diff
@@ -40,49 +40,36 @@ const Loader = () => (
+        return last ? (
+          <ActiveSegment key={urlSegment}>{name}</ActiveSegment>
+        ) : (
+          <Link to={makeEntityLink(urlSegment)} key={urlSegment}>
```
This is the last entity specific part of breadcrumbs - might as well get rid of it now while you're at it?

tcaiger replied:
@edmofro - done
